### PR TITLE
[py2py3] Migration at level scr/python/A/B/C - slice 15

### LINKED ISSUE #10139: [py2py3] Migration at level scr/python/A/B/C - slice 15
#### Intro

Our purpose is to continue with the gradual migration. 
This is the 15-th slice in which the migration has been divided into. 
We have split migration units at level `src/python/A/B/C`.

A detailed plan of the migration at this depth is available 
at the dmwm mkdocs [detailed plan page](https://cms-dmwm-test.docs.cern.ch/py2py3/plan_detailed/), 
with summary information on top.

#### Impact of the new feature

We start migrating the 15-th bulk of migration units 
at src/python/A/B/C level.

 ```plaintext
                       name  nfiles  gradual  future_pm
WMCore_MicroService_Unified      14    False        101
```

#### Is your feature request related to a problem? Please describe.

The end goal is to have WMCore py2 and py3 compatible.

#### Describe the solution you'd like

Run futurize on the first bulk of units 
and then apply the necessary manual changes.

### PR DESCRIPTION
**Pull request by mapellidario**
Fixes #10139 

#### Status
Ready

#### Related PRs

* Previous migration PR: #10313  (slice14, issue #10138 )
* Next migration PR: #10321 (slice16, issue #10140  )

#### Description

Run futurize and some manual changes on the first batch of src/python/A/B/C.

Notable changes:
* `src/python/WMCore/MicroService/Unified/TaskManager.py`: `str()` is used both for formatting error/exception messages and for converting to a type that can be used as argument for `hashlib.md5.update()`. This required encoding unicode strings to bytes and can make a `try-except` obsolete. Cleaning tracked in #10322 
* `src/python/WMCore/MicroService/Unified/RequestInfo.py`: used "native string" approach and did not imoprt str from builtins. This is unlikely to be problematic, but it may be useful to point out for future-me

`pylint --py3k`: the report warn us about the use of `round`, but it is unlikely that it is going to create problems for us. I consider the two warnings as false positives and I am happy with that report! :)

#### Is it backward compatible (if not, which system it affects?)

It should be. (Any possible cause for errors will we reported here)

#### External dependencies / deployment changes

Requires python-future in both py2 and py3 environments.

### CODE REVIEW

**Comment by mapellidario** [comment]
Test this please

File: src/python/WMCore/MicroService/Unified/MSTransferor.py [hunk 1]
```diff
@@ -416,7 +417,7 @@ def checkPUDataLocation(self, wflow):
 
         if secondaryAAA:
             # what matters is to have pileup dataset(s) available in ANY disk storage
-            for dset, dsetDict in pileupInput.items():
+            for dset, dsetDict in veiwitems(pileupInput):
```
Comment (amaltaro) [inline src/python/WMCore/MicroService/Unified/MSTransferor.py:420]: typo `veiwitems`

File: src/python/WMCore/MicroService/Unified/MSTransferor.py [hunk 2]
```diff
@@ -428,7 +429,7 @@ def checkPUDataLocation(self, wflow):
                     self.logger.info("secondary dataset %s not available even through AAA", dset)
         else:
             if len(pileupInput) == 1:
-                for dset, dsetDict in pileupInput.items():
+                for dset, dsetDict in veiwitems(pileupInput):
```
Comment (amaltaro) [inline src/python/WMCore/MicroService/Unified/MSTransferor.py:432]: another typo `veiwitems`

File: src/python/WMCore/MicroService/Unified/TaskManager.py [hunk 3]
```diff
@@ -28,9 +31,11 @@ def genkey(query):
         record = dict(query)
         query = json.JSONEncoder(sort_keys=True).encode(record)
     keyhash = hashlib.md5()
+    quey = encodeUnicodeToBytes(query)
```
Comment (amaltaro) [inline src/python/WMCore/MicroService/Unified/TaskManager.py:34]: I assume this `quey` is a typo as well?

File: src/python/WMCore/MicroService/Unified/TaskManager.py [hunk 4]
```diff
@@ -83,7 +88,7 @@ def add(self, uid):
         "Add given uid or increment uid occurence in a set"
         if  not uid:
             return
-        if  uid in self.set.keys():
+        if  uid in list(self.set.keys()):
```
Comment (amaltaro) [inline src/python/WMCore/MicroService/Unified/TaskManager.py:91]: Should we change it to `list(self.set)`?

**Review by amaltaro — CHANGES_REQUESTED** [review]
Dario, please consider a few comments made along the code.

Regarding the `round()` py3k lint complaint, I learned that it works in both py2/py3, but they can yield different results and different data type as well. I've done a couple of checks in py2/py3 and I see no issues with those 2 warnings from the Common module. Just wanted to highlight it though that it might be a problem for the job splitting modules.

File: src/python/WMCore/MicroService/Unified/MSTransferor.py (hunk 1 above)
Comment (mapellidario) [reply src/python/WMCore/MicroService/Unified/MSTransferor.py:420]: oh my god, this has been a terrible slice! i made a tooon of typos! i am so sorry!

**Review by amaltaro — APPROVED** [review]
It looks good to go, Dario. Can you please squash those commits? Thanks

**Pull request merged**